core/time_handling.py

```diff
@@ -2156,3 +2167,151 @@ def find_ak_ext_ana_precip_neighbors(supplemental_precip, config_options, d_curr
+
+
+def find_hourly_nbm_apcp_neighbors(supplemental_precip, config_options, d_current, mpi_config):
+    """
+    Function to calculate the previous and next NBM/CORE/CONUS files. This
+    will also calculate the neighboring radar quality index (RQI) files as well.
+    :param supplemental_precip:
+    :param config_options:
+    :param d_current:
+    :param mpi_config:
+    :return:
+    """
+    nbm_out_freq = {
+    36: 60,
+    240: 360
+    }
+        
+    # First we need to find the nearest previous and next hour, which is
+    # the previous/next NBM files we will be using.
+    current_yr = d_current.year
+    current_mo = d_current.month
+    current_day = d_current.day
+    current_hr = d_current.hour
+    current_min = d_current.minute
+ 
+    # First find the current NBM forecast cycle that we are using.
+    current_nbm_cycle = config_options.current_fcst_cycle - \
+        datetime.timedelta(seconds=supplemental_precip.userCycleOffset * 60.0)
+
+    # Calculate the current forecast hour within this NBM cycle.
+    dt_tmp = d_current - current_nbm_cycle
+    current_nbm_hour = int(dt_tmp.days*24) + int(dt_tmp.seconds/3600.0)
+
+    # Set the input file frequency to be hourly for f001-f036 and 6-hourly beyond f036.
+    if current_nbm_hour <= 36:
+        supplemental_precip.input_frequency = 60.0
+    else:
+        supplemental_precip.input_frequency = 360.0
+            
+    # Calculate the previous file to process.
+    min_since_last_output = (current_nbm_hour*60) % supplemental_precip.input_frequency
+    if min_since_last_output == 0:
+        min_since_last_output = supplemental_precip.input_frequency
+    prev_nbm_date = d_current - datetime.timedelta(seconds=min_since_last_output*60)
+    supplemental_precip.fcst_date1 = prev_nbm_date
+    if min_since_last_output == supplemental_precip.input_frequency:
+        min_until_next_output = 0
+    else:
+        min_until_next_output = supplemental_precip.input_frequency - min_since_last_output
+    next_nbm_date = d_current + datetime.timedelta(seconds=min_until_next_output * 60)
+    supplemental_precip.fcst_date2 = next_nbm_date
+
+    # Calculate the output forecast hours needed based on the prev/next dates.
+    dt_tmp = next_nbm_date - current_nbm_cycle
+    next_nbm_forecast_hour = int(dt_tmp.days*24.0) + int(dt_tmp.seconds/3600.0)
+    supplemental_precip.fcst_hour2 = next_nbm_forecast_hour
+    dt_tmp = prev_nbm_date - current_nbm_cycle
+    prev_nbm_forecast_hour = int(dt_tmp.days*24.0) + int(dt_tmp.seconds/3600.0)
+    supplemental_precip.fcst_hour1 = prev_nbm_forecast_hour
+    # If we are on the first NBM forecast hour (1), and we have calculated the previous forecast
+    # hour to be 0, simply set both hours to be 1. Hour 0 will not produce the fields we need, and
+    # no interpolation is required.
+    if prev_nbm_forecast_hour == 0:
+        prev_nbm_forecast_hour = 1
+
+    # Calculate expected file paths.
+    if supplemental_precip.keyValue == 8:
```

Comment:
It seems like the concept of tmp_file1 being the previous file and tmp_file2 being the current are flipped in this case from the paradigm used for the HRRR and some of the other inputs. This may be worth reviewing further.

For example, for the CONUS HRRR
    # Calculate expected file paths.
    tmp_file1 = input_forcings.inDir + '/hrrr.' + current_hrrr_cycle.strftime(
        '%Y%m%d') + "/conus/hrrr.t" + current_hrrr_cycle.strftime('%H') + 'z.wrfsfcf' + \
        str(prev_hrrr_forecast_hour).zfill(2) + input_forcings.file_ext
    if mpi_config.rank == 0:
        config_options.statusMsg = "Previous HRRR file being used: " + tmp_file1
        err_handler.log_msg(config_options, mpi_config)

    tmp_file2 = input_forcings.inDir + '/hrrr.' + current_hrrr_cycle.strftime(
        '%Y%m%d') + "/conus/hrrr.t" + current_hrrr_cycle.strftime('%H') + 'z.wrfsfcf' \
        + str(next_hrrr_forecast_hour).zfill(2) + input_forcings.file_ext
    if mpi_config.rank == 0:
        if mpi_config.rank == 0:
            config_options.statusMsg = "Next HRRR file being used: " + tmp_file2
            err_handler.log_msg(config_options, mpi_config)
    err_handler.check_program_status(config_options, mpi_config)

Reply:
The reality is, because temporal interpolation was never fully implemented, that in practice it doesn't really matter. But given that file_2 is the one typically used in the regridders, I'm hesitant to flip it back now due to unforeseen side-effects. When the regridders are refactored in the upcoming version, we can probably eliminate this entire section of code.